miner.go

```diff
@@ -28,10 +32,62 @@ type Miner struct {
 	needsWorkRefresh chan struct{}
 	wg               sync.WaitGroup
 	pool             *stratum.Stratum
+
+	rpc *rpcclient.Client
 }
 
-func NewMiner() (*Miner, error) {
+func NewMiner(ctx context.Context) (*Miner, error) {
+	ntfnHandlers := rpcclient.NotificationHandlers{
+		OnBlockConnected: func(blockHeader []byte, transactions [][]byte) {
+			minrLog.Infof("Block connected: %x (%d transactions)", blockHeader, len(transactions))
+		},
+		OnBlockDisconnected: func(blockHeader []byte) {
+			minrLog.Infof("Block disconnected: %x", blockHeader)
+		},
+		OnWork: func(data, target []byte, reason string) {
+			minrLog.Infof("Work received: %x %x %s", data, target, reason)
+
+			bigTarget := new(big.Int)
+			bigTarget.SetBytes(util.Reverse(target))
+
+			var workData [192]byte
+			copy(workData[:], data)
+			ts := workData[128+4*work.TimestampWord : 132+4*work.TimestampWord]
+			givenTs := binary.LittleEndian.Uint32(ts)
+			w := work.NewWork(workData, bigTarget, givenTs, uint32(time.Now().Unix()), true)
+
+			w.Target = bigTarget
+
+			// Solo
+			for _, d := range m.devices {
+				d.SetWork(ctx, w)
+			}
+		},
+	}
+	// Connect to local dcrd RPC server using websockets.
+	certs, err := os.ReadFile(cfg.RPCCert)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read rpc certificate %v: %w",
+			cfg.RPCCert, err)
+	}
+
+	connCfg := &rpcclient.ConnConfig{
+		Host:         cfg.RPCServer,
+		Endpoint:     "ws",
+		User:         cfg.RPCUser,
+		Pass:         cfg.RPCPassword,
+		Certificates: certs,
+		Proxy:        cfg.Proxy,
+		ProxyUser:    cfg.ProxyUser,
+		ProxyPass:    cfg.ProxyPass,
+	}
+	rpc, err := rpcclient.New(connCfg, &ntfnHandlers)
```

Comment:
This looks like it's going to require dcrd when stratum mining which isn't correct.  It can't depend on dcrd at all when doing pool mining via stratum.